lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb

```diff
@@ -402,7 +402,14 @@ def validate_after_scope_change?
           else
             all_records = @subject.class.all
             @options[:scopes].all? do |scope|
-              previous_value = all_records.map(&scope).compact.max
+              previous_value =
+                if is_a_boolean_scope?(all_records, scope)
+                  value = all_records.map(&scope).last
+                  @existing_record.destroy
```

Comment:
Why do we need to destroy the record here? I would think we'd want to keep it around, that way we're properly testing the validation.